hashicorp/boundary-ui
# fix: 🐛 Desktop yarn start:desktop build fix

## Description

[bug] - after updating `desktop` UI to Ember 4.12.2, the `yarn start:desktop` build process was erroring out with a reference to the `instantiate Watcher with (await Watcher.build()).watcher, not new Watcher()` error message.  After searching we found out that between Ember 4.4.1 and 4.12.2 they changed the way builds were imported in ember-cli.

https://github.com/ember-cli/ember-cli/blob/master/lib/models/watcher.js

[fix] - upgrade `ember-electron` to 4.12.2 to match the Watcher API

https://github.com/adopted-ember-addons/ember-electron/blob/main/lib/tasks/electron.js

## Screenshots (if appropriate):
### Before updating ember-electron

<img width="1011" alt="image" src="https://github.com/hashicorp/boundary-ui/assets/24277002/07d2eb83-3c68-4640-85d7-8375b0ff02b3">

### After updating ember-electron
<img width="1020" alt="Screenshot 2023-11-16 at 13 54 25" src="https://github.com/hashicorp/boundary-ui/assets/24277002/6de97bbc-5cbc-4aed-9048-f6d17e629d0c">
<img width="1257" alt="Screenshot 2023-11-16 at 13 54 40" src="https://github.com/hashicorp/boundary-ui/assets/24277002/af4993d9-82bf-46c1-9394-f7a6959dd511">

## How to Test

- [] cd into `ui/desktop' and run `yarn start:desktop` against Mirage or use `ENABLE_MRAG=false yarn start:desktop, if you have a `boundary dev` instance running
- desktop app should start up as normal

## Checklist:


~~- [ ] I have added JSON response output for API changes~~
~~- [ ] I have commented on my code, particularly in hard-to-understand areas~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
